Zend/Optimizer/zend_optimizer.c

```diff
@@ -1238,6 +1238,8 @@ static void zend_redo_pass_two_ex(zend_op_array *op_array, zend_ssa *ssa)
 	}
 #endif
 
+	op_array->T += ZEND_OBSERVER_ENABLED; // reserve last temporary for observers if enabled
+
```

Comment:
I don't understand why you decrement and increment ``op_array->T`` back and force.
It may be modified only by two Optimizer passes - ``optimize_temp_vars_5.c`` and ``compact_vars.c``.
Most probably these passes need to be updated to take OBSERVER into account.

Reply:
I agree. I wasn't really aware that only these two passes touch it - that makes much more sense then :-)